feature/experimental/isis/ate_tests/base_adjacencies_test/base_adjacencies_test.go

```diff
@@ -15,228 +15,313 @@
+// for this particular query.
+func EqualToDefault[T any](query ygnmi.SingletonQuery[T], val T) check.Validator {
+	if *deviations.AllowNilForDefaults {
+		return check.EqualOrNil(query, val)
+	}
+	return check.Equal(query, val)
 }
 
 // TestBasic configures IS-IS on the DUT and confirms that the various values and defaults propagate
 // then configures the ATE as well, waits for the adjacency to form, and checks that numerous
 // counters and other values now have sensible values.
 func TestBasic(t *testing.T) {
-	ts := session.NewWithISIS(t)
+	ts := session.MustNew(t).WithISIS()
 	// Only push DUT config - no adjacency established yet
-	ts.PushDUT(t)
-	isisRoot := ts.DUTISISTelemetry(t)
-	isisRoot.Global().Instance().Await(t, time.Second, session.ISISName)
+	if err := ts.PushDUT(context.Background()); err != nil {
+		t.Fatalf("Unable to push initial DUT config: %v", err)
+	}
+	isisRoot := session.ISISPath()
+	port1ISIS := isisRoot.Interface(ts.DUTPort1.Name())
+	if err := check.Equal(isisRoot.Global().Instance().State(), session.ISISName).AwaitFor(time.Second, ts.DUTClient); err != nil {
+		t.Fatalf("IS-IS failed to configure: %v", err)
+	}
+	// There might be lag between when the instance name is set and when the
+	// other parameters are set; we expect the total lag to be under 5s
+	deadline := time.Now().Add(time.Second * 5)
 
 	t.Run("read_config", func(t *testing.T) {
-		assert.Value(t, isisRoot.Global().Net(), []string{"49.0001.1920.0000.2001.00"})
-		assert.Value(t, isisRoot.Global().LevelCapability(), telemetry.IsisTypes_LevelType_LEVEL_1_2)
-		assert.Value(t, isisRoot.Global().Af(telemetry.IsisTypes_AFI_TYPE_IPV4, telemetry.IsisTypes_SAFI_TYPE_UNICAST).Enabled(), true)
-		assert.Value(t, isisRoot.Global().Af(telemetry.IsisTypes_AFI_TYPE_IPV6, telemetry.IsisTypes_SAFI_TYPE_UNICAST).Enabled(), true)
-		assert.Value(t, isisRoot.Level(2).Enabled(), true)
-		assert.Value(t, isisRoot.Interface(ts.DUT.Port(t, "port1").Name()).Enabled(), true)
-		assert.Value(t, isisRoot.Interface(ts.DUT.Port(t, "port1").Name()).CircuitType(), telemetry.IsisTypes_CircuitType_POINT_TO_POINT)
+		for _, vd := range []check.Validator{
+			check.Equal(isisRoot.Global().Net().State(), []string{"49.0001.1920.0000.2001.00"}),
+			EqualToDefault(isisRoot.Global().LevelCapability().State(), oc.Isis_LevelType_LEVEL_1_2),
+			check.Equal(isisRoot.Global().Af(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled().State(), true),
+			check.Equal(isisRoot.Global().Af(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled().State(), true),
+			check.Equal(isisRoot.Level(2).Enabled().State(), true),
+			check.Equal(port1ISIS.Enabled().State(), true),
+			check.Equal(port1ISIS.CircuitType().State(), oc.Isis_CircuitType_POINT_TO_POINT),
+		} {
+			t.Run(vd.RelPath(isisRoot), func(t *testing.T) {
+				if err := vd.AwaitUntil(deadline, ts.DUTClient); err != nil {
+					t.Error(err)
```

Comment:
Added vd.AwaitUntil(deadline, ts.DUTClient) to the error msg.

Reply:
I don't think it belongs in the error message - these checks are all designed to produce readable error messages like

```
/network-instances/network-instance[name=default]/protocols/protocol[identifier=ISIS][name=DEFAULT]/isis/level/state/enabled: got no value, want true
```

From the point of view of anyone reading the test output, this is the information they want, not the fact that it was part of an `AwaitUntil` call specifically. (I know Go testing style guidelines recommend putting the function being called in the error message, but that only makes sense if the function being called is the thing you're testing - we aren't testing `AwaitUntil` here, we're testing that this particular path has the correct value).